nixos/modules/services/misc/rkvm.nix

```diff
@@ -0,0 +1,168 @@
+                type = types.path;
+                default = "/etc/rkvm/certificate.pem";
+                description = mdDoc ''
+                  TLS certificate path. This should be generated with the rkvm-certificate-gen CLI.
+                '';
+              };
+
+              key = mkOption {
+                type = types.path;
+                default = "/etc/rkvm/key.pem";
+                description = mdDoc ''
+                  TLS key path.
+                '';
+              };
+
+              password = mkOption {
+                type = types.str;
+                description = mdDoc ''
+                  Shared secret token to authenticate the client.
+                  Make sure this matches your client's config.
+                '';
+              };
+            };
+          };
+
+        default = { };
+        description = mdDoc "Structured server dameon configuration";
+      };
+    };
+
+    client = {
+      enable = mkEnableOption "rkvm client daemon (input receiver)";
+
+      settings = mkOption rec {
+        type = types.submodule
+          {
+            freeformType = toml.type;
+            options = {
+              server = mkOption {
+                type = types.str;
+                example = "other:5258";
+                description = mdDoc ''
+                  Network address to connect to including a port.
+                  On *NIX you can specify an absolute path to connect to a UNIX domain socket.
+                '';
+              };
+
+              certificate = mkOption {
+                type = types.path;
+                default = "/etc/rkvm/certificate.pem";
+                description = mdDoc ''
+                  TLS ceritficate path.
+                '';
+              };
+
+              key = mkOption {
+                type = types.path;
+                default = "/etc/rkvm/key.pem";
+                description = mdDoc ''
+                  TLS key path.
+                '';
+              };
+
+              password = mkOption {
+                type = types.str;
+                description = mdDoc ''
+                  Shared secret token to authenticate the client.
+                  Make sure this matches your client's config.
+                '';
+              };
+            };
+          };
+
+        default = {
+          certificate = "/etc/rkvm/certificate.pem";
+        };
+        example = literalExpression ''{
+          certificate = "/etc/rkvm/certificate.pem";
+          server = "server.local:5258"
+        }'';
+        description = mdDoc "Structured client dameon configuration";
+      };
+    };
+
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    systemd.services =
+      let
+        mkBase = component: rec {
+          description = "RKVM ${component}";
+          wantedBy = [ "multi-user.target" ];
+          after = [ "network-online.target" ];
+          wants = after;
```

Comment:
The server and client services have different dependencies for network(-online).target.

Check my upstream fix [PR](https://github.com/htrefil/rkvm/pull/48) for more info.

Reply:
very nice catch! thank you (: